samples/Mvc.LocalizationSample.Web/CustomValidationAttributeAdapterProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.DataAnnotations;
+using Microsoft.AspNetCore.Mvc.DataAnnotations.Internal;
+using Microsoft.Extensions.Localization;
+
+namespace Mvc.LocalizationSample.Web
+{
+    public class CustomValidationAttributeAdapterProvider : IValidationAttributeAdapterProvider
+    {
+        public IAttributeAdapter GetAttributeAdapter(ValidationAttribute attribute, IStringLocalizer stringLocalizer)
+        {
+            var defaultProvider = new ValidationAttributeAdapterProvider();
```

Comment:
This approach couples the replacement provider with MVC `.Internal` implementation details. Suggest instead registering a `Func` which calls `serviceProvider.GetRequiredService<IValidationAttributeAdapterProvider>()` and passes that to a new instance of this provider.